base/bucket.go

```diff
@@ -188,12 +200,32 @@ func (spec *BucketSpec) GetGoCBConnString(params *GoCBConnStringParams) (string,
 	asValues := url.Values(connSpec.Options)
 
 	// Add kv_pool_size as used in both GoCB versions
-	poolSizeFromConnStr := asValues.Get("kv_pool_size")
-	if poolSizeFromConnStr == "" {
+	fromConnStr := asValues.Get("kv_pool_size")
+	if fromConnStr == "" {
 		asValues.Set("kv_pool_size", strconv.Itoa(params.KVPoolSize))
 		spec.KvPoolSize = params.KVPoolSize
 	} else {
-		spec.KvPoolSize, _ = strconv.Atoi(poolSizeFromConnStr)
+		spec.KvPoolSize, _ = strconv.Atoi(fromConnStr)
+	}
+
+	fromConnStr = asValues.Get("kv_buffer_size")
+	if fromConnStr == "" && spec.KvBufferSize != 0 {
+		asValues.Set("kv_buffer_size", strconv.Itoa(spec.KvBufferSize))
+	} else if fromConnStr == "" {
+		asValues.Set("kv_buffer_size", strconv.Itoa(params.KVBufferSize))
```

Comment:
What's the scenario where spec.KvBufferSize and params.KvBufferSize are different values?

Reply:
If you set the values through unsupported config on the database, the params values here above will still be set to the default serverless ones. If we set these values through the unsuppotrted options, the spec values on the bucket will be set immediately and should be different to the default serverless values. So ultimately this check here is to see if we have already set the buffers through the database config.

Reply:
I still don't understand why we have two different ways to specify this information (spec and params).  Are there other use cases that require the use of params?

Reply:
So for supporting changing the buffer sizes through the unsupported options api, inside my changes in server context I do a check to pull these values from the unsupported option if specified. If it is specified I add it to the bucket spec there. But after that function is called `GetBucketSpec` we call the `GetGoCBConnString` where we pass the default serverless values in. So I added these checks here to check if the spec values had already been set. If they have already been set its due to the unsupported options config and we don't want to override this with the default values being set in params. In contrast if no unsupported options are specified we need to set the string to the default serverless values. 
Is this the right approach or do you think I should just append these values onto the `server` string on the spec in `GetBucketSpec` and then In the `GetGoCBConnString` these values wouldn't be overwritten as they have been set already?

Reply:
I think it would be preferable to make the 'unsupported vs default' evaluation in _getOrAddDatabaseFromConfig, and use that to set the values of GoCBConnStringParams there.  That will make the behaviour of this function more understandable and consistent with what we do for kv pool size.